docs/tutorial/bit_extraction.md

```diff
@@ -0,0 +1,122 @@
+# Bit Extraction
+
+Some applications require directly manipulating bits of integers. Concrete provides bit extraction operation for such applications.
+
+{% hint style="warning" %}
+Bit extraction only works in `Native` encoding, which is usually selected when all table lookups in the circuit are below or equal to 8 bits.
+{% endhint %}
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return fhe.bits(x)[0], fhe.bits(x)[3]
+
+inputset = range(32)
+circuit = f.compile(inputset)
+
+assert circuit.encrypt_run_decrypt(0b_00000) == (0, 0)
+assert circuit.encrypt_run_decrypt(0b_00001) == (1, 0)
+
+assert circuit.encrypt_run_decrypt(0b_01100) == (0, 1)
+assert circuit.encrypt_run_decrypt(0b_01101) == (1, 1)
+```
+
+Slices can be used for indexing `fhe.bits(value)` as well.
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return fhe.bits(x)[1:4]
+
+inputset = range(32)
+circuit = f.compile(inputset)
+
+assert circuit.encrypt_run_decrypt(0b_01101) == 0b_110
+assert circuit.encrypt_run_decrypt(0b_01011) == 0b_101
+```
+
+Even slices with negative steps are supported!
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return fhe.bits(x)[3:0:-1]
```

Comment:
how do i get in reverse order including bit 0 ?
ValueError: Extracting bits without an upper bound (stop is None) isn't supported on signed values (e.g., [-2 -4])

Reply:
I don't think you can using this. but we might be able to make an exception if the step is negative, we could allow None in the stop position maybe :thinking:

Reply:
I see no reason why 3:-1:-1  would be not supported, it's just the reverse of 0:4 which is supported.

Reply:
That wouldn't work as `-1` means from `3` to `first element from the back` in python.

However, I found this https://stackoverflow.com/a/49827561 and I'll add support for it :+1: